Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -64,10 +65,10 @@ template <class Type> class TableColumn : public API::Column {
     std::string name = std::string(typeid(Type).name());
     if ((name.find('i') != std::string::npos) || (name.find('l') != std::string::npos) ||
         (name.find('x') != std::string::npos)) {
-      if (length == 4) {
```

Comment:
There is a lot of work we could do to make TableColumn not so ugly. Example: move implementation into the cpp file. That is a job for a different PR.